Automattic/jetpack
# Launchpad: Add "Write 3 posts" task for Newsletter

Fixes https://github.com/Automattic/wp-calypso/issues/79612

## Proposed changes:

Adds the "Write 3 posts" task for the Newsletter flows. This task will implement the usage of the count properties added in the PR linked below.

**Note**: This PR depends on https://github.com/Automattic/jetpack/pull/31970 

More context:  p1689702956055999-slack-C057AH42XQD

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Make sure the public API is sandboxed
* Add the following code to your 0-sandbox.php file to enable the Newsletters checklists:
```
add_filter( 'wpcom_launchpad_intent_free_newsletter_enabled', '__return_true' );
```
* Apply this PR to your sandbox using the comment below: https://github.com/Automattic/jetpack/pull/32012#issuecomment-1645584329
* Navigate to `/setup/newsletter/intro` and create a Free Newsletter.
* Open the developer console: https://developer.wordpress.com/docs/api/console/
* Make a `GET` request to `WP REST API` > `wpcom/v2` > /sites/:siteSlug/launchpad?_locale=user&checklist_slug=intent-free-newsletter&_envelope=1
* The response should contain the `Write 3 posts` task with the following props:
  - target_repetitions: 3
  - repetition_count: 0
 * Write a new post on your new site
 * And make the request again. The repetition_count should now be 1.

![Screen Shot 2023-07-25 at 10 22 08](https://github.com/Automattic/jetpack/assets/1234758/639d8aa3-ec28-455a-9e06-4824a4e1a6bb)
